field/track.go

```diff
@@ -0,0 +1,404 @@
+package field
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"regexp/syntax"
+	"strings"
+	"time"
+
+	"github.com/moov-io/iso8583/utils"
+)
+
+var _ Field = (*Track)(nil)
+
+type Track struct {
+	spec *Spec  `xml:"-" json:"-"`
+	data *Track `xml:"-" json:"-"`
+
+	// Available for Track 1
+	FixedLength bool `xml:"-" json:"-"`
+
+	// Available versions 1,2,3
+	Version TrackVersion `xml:"Version,omitempty" json:"version,omitempty"`
+
+	// Available for Track 1, Track 3
+	FormatCode string `xml:"FormatCode,omitempty" json:"format_code,omitempty"`
+
+	// Available for Track 1, Track 2, Track 3
+	PrimaryAccountNumber string `xml:"PrimaryAccountNumber,omitempty" json:"primary_account_number,omitempty"`
+
+	// Available for Track 1
+	Name string `xml:"Name,omitempty" json:"name,omitempty"`
+
+	// Available for Track 1, Track 2
+	ExpirationDate *time.Time `xml:"ExpirationDate,omitempty" json:"expiration_date,omitempty"`
+
+	// Available for Track 1, Track 2
+	ServiceCode string `xml:"ServiceCode,omitempty" json:"service_code,omitempty"`
+
+	// Available for Track 1, Track 2, Track 3
+	//  If track 3, the field describe security data + additional data
+	DiscretionaryData string `xml:"DiscretionaryData,omitempty" json:"discretionary_data,omitempty"`
+}
+
+type TrackVersion int
+
+func (v TrackVersion) Valid() bool {
+	if v == VersionFirst || v == VersionSecond || v == VersionThird {
+		return true
+	}
+	return false
+}
+
+const (
+	VersionFirst  = 1
+	VersionSecond = 2
+	VersionThird  = 3
+)
```

Comment:
If these are to be of type `TrackVersion` lets define them so, as currently they are `int`s
```suggestion
type TrackNumber int

const (
    Track1 TrackNumber = 1
    Track2 TrackNumber = 2
    Track3 TrackNumber = 3
)
```
Within this suggestion I also renamed the variables as `field.VersionFirst` to me doesn't sound related to track data.
What do you think?